include/openssl/ssl.h.in

```diff
@@ -2258,6 +2258,20 @@ __owur int SSL_set_blocking_mode(SSL *s, int blocking);
 __owur int SSL_get_blocking_mode(SSL *s);
 __owur int SSL_set_initial_peer_addr(SSL *s, const BIO_ADDR *peer_addr);
 
+typedef struct ssl_shutdown_ex_args_st {
+    uint64_t    quic_error_code;
+    const char  *quic_reason;
+} SSL_SHUTDOWN_EX_ARGS;
+
+#define SSL_SHUTDOWN_FLAG_RAPID         (1U << 0)
+#define SSL_SHUTDOWN_FLAG_IMMEDIATE     (1U << 1)
```

Comment:
The flag names can be confused with each other. I could see something as SSL_SHUTDOWN_FLAG_NO_STREAM_FLUSH instead of SSL_SHUTDOWN_FLAG_IMMEDIATE to clarify things.